syntax = "proto2";

message CountryEmpty {}

message States {
	required string id = 1;
}

message CountryRequest {
	required string id = 1;
	required string name = 2;
	required string phone_prefix = 3;
	optional bool active = 4 [default = true];
	repeated States states = 5;
}

message CountryMultipleResponse {
	repeated CountryRequest country = 1;
}

service Country {
	rpc get_all (CountryEmpty) returns (CountryMultipleResponse);
}